homeassistant/components/plugwise/sensor.py

```diff
@@ -0,0 +1,434 @@
+                            device_properties["name"],
+                            dev_id,
+                            sensor,
+                            sensor_type,
+                        )
+                    )
+                _LOGGER.info("Added sensor %s", device_properties["name"])
+
+        if single_thermostat is False:
+            for state in INDICATE_ACTIVE_LOCAL_DEVICE:
+                if state in data:
+                    entities.append(
+                        PwAuxDeviceSensor(
+                            api,
+                            coordinator,
+                            device_properties["name"],
+                            dev_id,
+                            DEVICE_STATE,
+                        )
+                    )
+                    _LOGGER.info("Added auxiliary sensor %s", device_properties["name"])
+                    break
+
+    async_add_entities(entities, True)
+
+
+class SmileSensor(SmileGateway):
+    """Represent Smile Sensors."""
+
+    def __init__(self, api, coordinator):
+        """Initialise the sensor."""
+        super().__init__(api, coordinator)
+
+        self._dev_class = None
+        self._state = None
+        self._unit_of_measurement = None
+
+    @property
+    def device_class(self):
+        """Device class of this entity."""
+        if not self._dev_class:
+            return None
+        return self._dev_class
+
+    @property
+    def state(self):
+        """Device class of this entity."""
+        if not self._state:
+            return None
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity, if any."""
+        if not self._unit_of_measurement:
+            return None
+        return self._unit_of_measurement
+
+
+class PwThermostatSensor(SmileSensor, Entity):
+    """Thermostat and climate sensor entities."""
+
+    def __init__(self, api, coordinator, name, dev_id, sensor, sensor_type):
+        """Set up the Plugwise API."""
+        super().__init__(api, coordinator)
+
+        self._api = api
+        self._gateway_id = self._api.gateway_id
+        self._dev_id = dev_id
+        self._sensor_type = sensor_type
+        self._entity_name = name
+        self._sensor = sensor
+
+        self._model = self._sensor_type[SENSOR_MAP_MODEL]
+        self._unit_of_measurement = self._sensor_type[SENSOR_MAP_UOM]
+        self._dev_class = self._sensor_type[SENSOR_MAP_DEVICE_CLASS]
+        self._icon = self._sensor_type[SENSOR_MAP_ICON]
+
+        if self._dev_id == self._api.heater_id:
+            self._entity_name = "Auxiliary"
+
+        sensorname = sensor.replace("_", " ").title()
+        self._name = f"{self._entity_name} {sensorname}"
+
+        if self._dev_id == self._api.gateway_id:
+            self._entity_name = f"Smile {self._entity_name}"
+
+        self._unique_id = f"{dev_id}-{sensor}"
+
+    def _process_data(self):
```

Comment:
Please name this `_async_*` and decorate it with `@callback` imported from core.py.